### ISSUE TITLE
Tuleap behind nginx reverse proxy leading to "Too many redirects"

### ISSUE THREAD

**Issue by lreiher** [Enalean/docker-tuleap-aio#65]
I am trying to run the Tuleap docker image using docker-compose behind [jwilder/nginx-proxy](https://github.com/jwilder/nginx-proxy). When trying to open up the Tuleap interface however, I am redirected too many times. I noticed that there have already been many issues regarding this kind of problem, but still couldn't figure out a solution.

Tuleap is running on its standard ports inside its container. It is on the same docker network with the reverse proxy. I added the following lines to the location part of the automatically generated nginx config for Tuleap:
```
proxy_pass http://tuleap.mydomain.com;
proxy_ssl_verify off;
proxy_set_header X-Real-IP         $remote_addr;
# Allow to know what is the original IP address (esp. for logging purpose as well a$
proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
# Allow to know what is the original protocol (so Tuleap knows if things were in HT$
proxy_set_header X-Forwarded-Proto $scheme;
# What is the name of the platform to the end users
proxy_set_header Host              $host;
# Write Destination header for Subversion COPY and MOVE operations
set $fixed_destination $http_destination;
if ( $http_destination ~* ^https(.*)$ ) {
    set $fixed_destination http$1;
}
proxy_set_header Destination $fixed_destination;
```
The `proxy_pass http://tuleap.mydomain.com;` is automatically generated by the reverse-proxy.

Additionally, after the first boot of the Tuleap container, I edited `/data/etc/tuleap/conf/local.inc` to set the trusted proxies setting to the Docker network, where both the reverse-proxy and Tuleap are in:
```
$sys_trusted_proxies = '172.18.0.0/16';
```

I also tried restarting the Tuleap container and the nginx inside the Tuleap container. Despite these configurations, the "too many redirects"-error still persists.

I guess I followed the steps from the [Tuleap Advanced Deployments Documentation](https://tuleap-documentation.readthedocs.io/en/latest/administration-guide/advanced-deployments.html) and now cannot figure out what else I could do. Do you guys have any ideas or see any problems with above configurations? Are there any log-files inside the Tuleap-container, that could help? Thanks in advance.

**Comment by LeSuisse** [comment]
Hi,

TL;DR: you probably need to set VIRTUAL_PROTO=https and VIRTUAL_PORT=443 to make it works. See https://github.com/jwilder/nginx-proxy#ssl-backends

According to the extract of the configuration you have given, you try to reach the Tuleap container with HTTP on 80/TCP. By default, the configuration generated by Tuleap will try to do everything to use HTTPS, one of the steps to enforce that is default nginx configuration that will just redirect everything to HTTPS (see https://github.com/Enalean/tuleap/blob/10.9/src/etc/nginx/tuleap.conf.dist#L35). It is likely that the root cause of your issue.

**Comment by lreiher** [comment]
Perfect, thank you so much for pointing this out. I didn't think about the fact that once connected to the nginx-proxy via HTTPS, the proxy itself makes HTTP contact to the Tuleap container. This solves the issue.
